src/lib/models/events-group.ts

```diff
@@ -53,11 +71,21 @@ export class EventsGroup {
         event?.activityTaskScheduledEventAttributes?.activityType?.name;
     }
 
-    if (isTimerStartedEvent(event)) {
+    if (isTimerEvent(event)) {
       this.id = event?.timerStartedEventAttributes?.timerId;
       this.name = `Timer ${event?.timerStartedEventAttributes?.timerId}`;
     }
 
+    if (isSignalEvent(event)) {
+      this.id = event?.id;
```

Comment:
You probably don't need `.?` on a single property. Either way, it will turn out to be `undefined`. It probably adds some extra weight after compilation since it _might_ be compiled down to checking if the property exists—unless the compiler is really smart.

Reply:
right